Addresses Issue #27

saaish opened the pull request.

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+* [Amazon EKS cluster](https://docs.aws.amazon.com/eks/latest/userguide/create-cluster.html) 
+* [AWS CLI version 2](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html)
+* [Terraform CLI](https://www.terraform.io/downloads)
+* [AWS Distro for Opentelemetry](https://aws.amazon.com/blogs/mt/amazon-managed-service-for-prometheus-is-now-generally-available/)
```
Opentelemetry -> OpenTelemetry and let's use https://aws-otel.github.io/ rather than linking to a blog post.

saaish replied:
modifications made as suggested

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+* [AWS Distro for Opentelemetry](https://aws.amazon.com/blogs/mt/amazon-managed-service-for-prometheus-is-now-generally-available/)
+* [eksctl](https://eksctl.io/)
+* [kubectl](https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html)
+* [jq](https://stedolan.github.io/jq/download/%20)
```
fix link

saaish replied:
Fixed the link

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+* [eksctl](https://eksctl.io/)
+* [kubectl](https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html)
+* [jq](https://stedolan.github.io/jq/download/%20)
+* [helm](https://www.eksworkshop.com/beginner/060_helm/helm_intro/install/index.html)
```
use https://helm.sh/ please

saaish replied:
Modifed

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+* [helm](https://www.eksworkshop.com/beginner/060_helm/helm_intro/install/index.html)
+* [SNS topic]
+
+In the recipe, we will use a sample application in order to demonstrate the metric scraping using ADOT and remote write the metrics to the workspace. Fork and clone the sample app from the repository at aws-otel-community (https://github.com/aws-observability/aws-otel-community). Execute the below commands to deploy application inside the EKS cluster. Make sure the terminal has access to the EKS  ...
```
First time mention ADOT, spell it out (or intro the abbreviation in pre-req section).

mhausenblas replied:
You can drop the last sentence "Make sure the terminal …"

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+* [kubectl](https://docs.aws.amazon.com/eks/latest/userguide/install-kubectl.html)
+* [jq](https://stedolan.github.io/jq/download/%20)
+* [helm](https://www.eksworkshop.com/beginner/060_helm/helm_intro/install/index.html)
+* [SNS topic]
```
link missing?

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+
+In the recipe, we will use a sample application in order to demonstrate the metric scraping using ADOT and remote write the metrics to the workspace. Fork and clone the sample app from the repository at aws-otel-community (https://github.com/aws-observability/aws-otel-community). Execute the below commands to deploy application inside the EKS cluster. Make sure the terminal has access to the EKS  ...
+
+This Prometheus sample app generates all 4 Prometheus metric types (counter, gauge, histogram, summary) and exposes them at the /metrics endpoint
```
4 -> four and `/metrics`

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+
+This Prometheus sample app generates all 4 Prometheus metric types (counter, gauge, histogram, summary) and exposes them at the /metrics endpoint
+
+A health check endpoint also exists at /
```
`/` rather than plain and can you connect it to the previous sentence pls?

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+
+A health check endpoint also exists at /
+
+The following is a list of optional command line flags for configuration:
```
Turn into a proper list, turn flags into code (for example `listen_address`)

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
````diff
@@ -0,0 +1,141 @@
+cd ./sample-apps/prometheus
+docker build . -t prometheus-sample-app:latest
+```
+2. Push this image to a registry such as Amazon ECR or DockerHub.
````
Let's focus on one, say, ECR and provide the commands. You can reuse, for example, https://aws-observability.github.io/aws-o11y-recipes/recipes/fargate-eks-metrics-go-adot-ampamg/#build-container-image

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
```diff
@@ -0,0 +1,141 @@
+
+### Deploying the Terraform module to configure Amazon Managed service for Prometheus workspace, recording rules & alert manager
+
+Now, we will  provision a Amazon Managed service for Prometheus workspace and will define an alerting rule that causes the Alert Manager to send a notification if a certain condition (defined in expr) holds true for a specified time period (for). Code in the Terraform language is stored in plain text files with the .tf file extension. There is also a JSON-based variant of the language that is nam ...
```
service -> Service

mhausenblas replied:
Turn code fragments such expr and for into code using `expr` and `for`

saaish replied:
changed

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
````diff
@@ -0,0 +1,141 @@
+}
+```
+
+Once the file is created, execute the below commands to provision the resource using terraform:
````
Turn above file into a proper resource (supporting file as per https://github.com/aws-observability/aws-o11y-recipes/#writing-recipes) and make sure to clean it up. It looks like there's an extra `EOF` in there?

saaish replied:
have created a file main.tf as per the direction

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
````diff
@@ -0,0 +1,141 @@
+terraform apply
+```
+
+Once the above steps are complete, verify the setup end-to-end by using awscurl and query the endpoint. Look for the metric “metric:recording_rule”, and, if you successfully find the metric, then you’ve successfully created a recording rule:
````
If you are using `awscurl` here you need to add it to the pre-reqs section as well.

saaish replied:
added the pre-req

mhausenblas commented on docs/recipes/amp-alertmanager-terraform.md:
````diff
@@ -0,0 +1,141 @@
+awscurl https://aps-workspaces.us-east-1.amazonaws.com/workspaces/$WORKSPACE_ID/alertmanager/api/v2/alerts --service="aps" -H "Content-Type: application/json"
+```
+
+## Clean up
````
Remove this section, it's a dupe (see below)

saaish replied:
Cleared the section

mhausenblas reviewed (changes requested):
Thanks for the recipe, some work required, left comments inline. Please address and re-submit.

mhausenblas commented:
OK, thanks for addressing the issues @saaish but there are still some things to be done before we can merge it:

* In `amp.md` you're linking to `prometheus-adapter-custom-metrics-autoscaling.md` where it should be `amp-alertmanager-terraform.md`
* You need to [locally preview](https://github.com/aws-observability/aws-o11y-recipes#local-preview) the rendering and you will then notice that a number of things are not properly formatted in Markdown, for example the pre-req list or the flags.
* Please make sure that all variables and files are properly formatted, for example `SNS_TOPIC` not SNS_TOPIC.
* For `main.tf` you need to call out all env vars that customers need to replace.
* Split the `awscurl` commands and show the output.
* The section headings should be short, ideally fitting in one line in the right-hand site ToC, for example, rather than "Deploying the Terraform module to configure Amazon Managed service for Prometheus workspace, recording rules & alert manager" say "Configure workspace with Terraform".

saaish commented:
> OK, thanks for addressing the issues @saaish but there are still some things to be done before we can merge it:
> 
> * In `amp.md` you're linking to `prometheus-adapter-custom-metrics-autoscaling.md` where it should be `amp-alertmanager-terraform.md`
> * You need to [locally preview](https://github.com/aws-observability/aws-o11y-recipes#local-preview) the rendering and you will then notice that a number of things are not properly formatted in Markdown, for example the pre-req list or the flags.
> * Please make sure that all variables and files are properly formatted, for example `SNS_TOPIC` not SNS_TOPIC.
> * For `main.tf` you need to call out all env vars that customers need to replace.
> * Split the `awscurl` commands and show the output.
> * The section headings should be short, ideally fitting in one line in the right-hand site ToC, for example, rather than "Deploying the Terraform module to configure Amazon Managed service for Prometheus workspace, recording rules & alert manager" say "Configure workspace with Terraform".

Thanks for the feedback @mhausenblas . I incorporated the changes and did a local preview. Things looked good. Please do the needful

mhausenblas commented:
@saaish thanks and there are still a number of things not addressed. From the flags (not a list, not code, see my previous comment) to the region env variable in `main.tf` you're not calling out to the inline comments above. At this point it might be faster if I merge it and take care of the edits myself rather than continue to play ping-pong here, lemme know what you prefer.
